pkg/safepath/safepath.go

```diff
@@ -214,6 +214,26 @@ func StatAtNoFollow(path *Path) (os.FileInfo, error) {
 	return os.Stat(pathFd.SafePath())
 }
 
+func GetxattrNoFollow(path *Path, attr string) ([]byte, error) {
+	var ret []byte
```

Comment:
Could you change this a little bit and let it directly use `unix.FGetxattr` instead of `Getxattr`?

I experienced on some kernels the issue that doing `Getxattr` on a filedescriptor path `/self/proc/fd/<num>` always returns `scp_t` even if the label is `container_t` already. It works consistently with this snippet:


```go
	// let's first find out the actual buffer size
	var buffer []byte
	labelLength, err := unix.Fgetxattr(int(file.Fd()), xattrNameSelinux, buffer)
	if err != nil {
		return "", fmt.Errorf("error reading fgetxattr: %v", err)
	}
	// now ask with the needed size
	buffer = make([]byte, labelLength)
	labelLength, err = unix.Fgetxattr(int(file.Fd()), xattrNameSelinux, buffer)
	if err != nil {
		return "", fmt.Errorf("error reading fgetxattr: %v", err)
	}
	if labelLength > 0 && buffer[labelLength-1] == '\x00' {
		labelLength = labelLength - 1
	}
	return string(buffer[:labelLength]), nil
```

The main issue is that it is this:

 * It is not always possible to relabel due to container runtime settings or due to the kernel
 * Most of the time the label is however already correct anyway
 * With this fix we read the label always correct, which limits failed writes soley to situations where we really need to change the label
 * And if the label does not match we have to fail, since `masquerade` networking would be broken.

Reply:
I couldn't get fgetxattr to work, I keep getting `error reading fgetxattr for root: /proc/21647/root, relative: /: bad file descriptor`
Please note that this PR is not about relabelling dev nodes, but rather finding out which categories the `virt-launcher` pod got to use them for `hp-volume`.

Reply:
@rmohr is this still a concern? Do you know what could be causing the error above with fgetxattr? Thanks

Reply:
@rmohr this is what I get when trying to use `Fgetxattr`: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8778/pull-kubevirt-e2e-k8s-1.25-sig-storage/1592178404596125696